### PR TITLE
fix(helm): set readOnlyRootFilesystem/runAsNonRoot, create a ServiceAccount in correct release namespace

### PR DESCRIPTION
**Pull request by michaelbeaumont**
`kube-linter` caught these. Everything with `helm` `install/upgrade/delete` appears to work.

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues -- #6049 
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: deployments/charts/kuma/templates/pre-upgrade-install-missing-crds-job.yaml [hunk 1]
```diff
@@ -113,15 +114,18 @@ spec:
...
       {{ toYaml .Values.hooks.podSecurityContext | trim | nindent 8 }}
       {{- end }}
       containers:
         - name: pre-upgrade-job
           image: {{ include "kubectl.formatImage" (dict "image" .Values.kubectl.image "root" $) | quote }}
-          {{- if .Values.hooks.containerSecurityContext }}
           securityContext:
+            runAsUser: 65534 # the nobody user in the default image
+            readOnlyRootFilesystem: true
+          {{- if .Values.hooks.containerSecurityContext }}
```
Comment (lahabana) [inline deployments/charts/kuma/templates/pre-upgrade-install-missing-crds-job.yaml:128]: Can you add a test that introduces a custom security context which overrides runAsUser for example? I feel like the override will create invalid yaml and you may need to use [merge](https://helm.sh/docs/chart_template_guide/function_list/#merge-mustmerge) instead

Comment (michaelbeaumont) [reply deployments/charts/kuma/templates/pre-upgrade-install-missing-crds-job.yaml]: done

File: app/kumactl/cmd/install/install_control_plane_test.go [hunk 2]
```diff
@@ -316,6 +316,14 @@ controlPlane:
...
 			goldenFile:  "install-control-plane.gateway-api-present-not-enabled.yaml",
 		}),
+		Entry("should work with --set", testCase{
+			extraArgs: []string{
+				"--set",
+				"controlPlane.podSecurityContext.runAsUser=65000,controlPlane.podSecurityContext.runAsNonRoot=false",
+			},
+			includeCRDs: true,
+			goldenFile:  "install-control-plane.with-pod-security.yaml",
+		}),
```
Comment (lahabana) [inline app/kumactl/cmd/install/install_control_plane_test.go:326]: If you want as it's a little easier you can have new tests by just adding values files in `app/kumactl/cmd/install/testdata/install-cp-helm` I find this way it's also easier to look at test cases

File: app/kumactl/cmd/install/testdata/install-control-plane.with-pod-security.yaml [hunk 3]
```diff
@@ -0,0 +1,5798 @@
...
+    app.kubernetes.io/name: kuma
+    app.kubernetes.io/instance: kuma
+data:
+  config.yaml: |
+    # use this file to override default configuration of `kuma-cp`
+    #
+    # see conf/kuma-cp.conf.yml for available settings
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
```
Comment (lahabana) [inline app/kumactl/cmd/install/testdata/install-control-plane.with-pod-security.yaml:36]: Also the `install-cp-helm` doesn't include the CRDs in the golden files. This is nice as it makes PRs that updates CRDs much smaller :)

**Review by lahabana — APPROVED** [review]
LGTM but I'd really move the new test in `install-cp-helm`

File: app/kumactl/cmd/install/install_control_plane_test.go (hunk 2 above)
Comment (michaelbeaumont) [reply app/kumactl/cmd/install/install_control_plane_test.go:326]: Will do!

**Pull request merged**